src/nomad_simulations/schema_packages/model_method.py

```diff
@@ -81,308 +91,661 @@ class ModelMethod(BaseModelMethod):
+    relativistic_correction = SubSection(
+        sub_section=RelativityModel.m_def,
+        repeats=False,
+        description="""
+        Relativistic treatment details.
         """,
     )
 
-    def __init__(self, m_def: 'Section' = None, m_context: 'Context' = None, **kwargs):
-        super().__init__(m_def, m_context, **kwargs)
-        self._jacobs_ladder_map = {
-            'lda': 'LDA',
-            'gga': 'GGA',
-            'mgg': 'meta-GGA',
-            'hyb_mgg': 'hyper-GGA',
-            'hyb': 'hybrid',
-        }
+    core_electron_treatment = SubSection(
+        sub_section=CoreElectronTreatment.m_def,
+        repeats=True,
+        description="""Core-electron approximation (ECP/PP) and any embedded relativistic character.
+        """,
+    )
 
-    def resolve_libxc_names(
-        self, xc_functionals: list[XCFunctional]
-    ) -> list[str] | None:
-        """
-        Resolves the `libxc_names` and sorts them from the list of `XCFunctional` sections.
 
-        Args:
-            xc_functionals (list[XCFunctional]): The list of `XCFunctional` sections.
+class XCComponent(ArchiveSection):
+    """
+    One XC component normalized to LibXC. Identity from the registry;
+    α/ω are ONLY set by parsers when present in inputs/outputs.
+    """
 
-        Returns:
-            (Optional[list[str]]): The resolved and sorted `libxc_names`.
-        """
-        return sorted(
-            [
-                functional.libxc_name
-                for functional in xc_functionals
-                if functional.libxc_name is not None
-            ]
-        )
+    # Identity (from LibXC registry)
+    libxc_id = Quantity(type=np.int64, description='LibXC ID (e.g., 101).')
+    canonical_label = Quantity(
+        type=str, description="LibXC label, e.g. 'XC_GGA_X_PBE'."
+    )
+    display_name = Quantity(type=str, description='Human-readable name from LibXC.')
 
-    def resolve_jacobs_ladder(
-        self,
-        libxc_names: list[str],
-    ) -> str:
-        """
-        Resolves the `jacobs_ladder` from the `libxc_names`. The mapping (libxc -> NOMAD) is set in `self._jacobs_ladder_map`.
+    # Taxonomy
+    family = Quantity(
+        type=MEnum('LDA', 'GGA', 'meta-GGA', 'hybrid-GGA', 'hybrid-meta-GGA'),
+        description="Jacob's ladder family.",
+    )
+    kind = Quantity(
+        type=MEnum('exchange', 'correlation', 'xc', 'k'), description='Component kind.'
+    )
 
-        Args:
-            libxc_names (list[str]): The list of `libxc_names`.
+    # Hybrid / range-separated parameters — PARSERS fill these if known
+    fraction_exact_exchange = Quantity(
+        type=np.float64, description='Global HF mixing α.'
+    )
+    sr_fraction_exact_exchange = Quantity(
+        type=np.float64, description='Short-range HF α_SR.'
+    )
+    lr_fraction_exact_exchange = Quantity(
+        type=np.float64, description='Long-range HF α_LR.'
+    )
```

Comment:
Most often, functionals have different terms (with potentially different exact exchange) for SR and LR, e.g. HSE yields ExcHSE=aExHF,SR(ω)+(1−a)ExPBE,SR(ω)+ExPBE,LR(ω)+EcPBE

Given that we are writing out terms, I would have a quantity that denotes the range specification.